lib/orderbook/OrderBook.ts

```diff
@@ -40,6 +48,9 @@ class OrderBook extends EventEmitter {
 
   private repository: OrderBookRepository;
 
+  /** Max matching retries for swap failures */
+  private static MAX_MATCHING_RETRIES = 3;
```

Comment:
I was actually more concerned about some problem with the swap clients which we won’t monitor well, that will cause all swaps to fail. In that case, delays isn’t the problem, so retries limit is better. 
Let me know if you think otherwise. 

In regards to the time limit, it’s still relevant for `executeSwap` (a single swap execution), but it will require an interrupt signal mechanism for the swaps, which I wasn’t sure what is our plan for.

Reply:
Ah I see - although maybe if one of our swap clients is non-functional, it wouldn't be so bad to error out the orders we can't fill anyway, effectively removing them from the order book.

I was thinking actually of more of a "soft" timeout here, as in if we exceed the limit we don't attempt a new swap but we also don't terminate ongoing swaps. I think we can deal with handling a long-running single swap separately, not as part of the order matching/placeorder logic.

Reply:
Ok, then i'll use here a timeout instead.